docs/strategies/circuit-breaker.md

```diff
@@ -99,3 +99,229 @@ await manualControl.CloseAsync();
+            .Handle<HttpRequestException>()
+            .Handle<BrokenCircuitException>(),
+        DelayGenerator = static args =>
+        {
+            _ = args.Context.Properties.TryGetValue(SleepDurationKey, out var delay);
+            delay ??= TimeSpan.FromSeconds(1);
+            return ValueTask.FromResult(delay);
+        }
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+**Reasoning**:
+- With this approach the two strategies are less coupled
+  - They both use the context and the `sleepDurationKey` components
+- The Circuit Breaker shares the `BreakDuration` via the context whenever it breaks
+  - When it transitions back to Closed then it "revokes" the sharring
+- The Retry do not have any circuit breaker specific logic
+  - It dynamically fetches the sleep duration whithout knowing anything about the circuit breaker
+  - Also if `BreakDuration` needs to be adjusted then it can be done in a single place
+
+### 2 - Using different duration for breaks
+
+In case of Retry you can specify dynamically the sleep duration via the `DelayGenerator`.
+In case of Circuit Breaker the `BreakDuration` is considered constant (can't be changed between breaks).
+
+❌ DON'T
+
+Use `Task.Delay` inside `OnOpened`
+
+<!-- snippet: circuit-breaker-anti-pattern-2 -->
+```cs
+static IEnumerable<TimeSpan> GetSleepDuration()
+{
+    for (int i = 1; i < 10; i++)
+    {
+        yield return TimeSpan.FromSeconds(i);
+    }
+}
+
+var sleepDurationProvider = GetSleepDuration().GetEnumerator();
+sleepDurationProvider.MoveNext();
+
+var circuitBreaker = new ResiliencePipelineBuilder()
+    .AddCircuitBreaker(new()
+    {
+        ShouldHandle = new PredicateBuilder().Handle<HttpRequestException>(),
+        BreakDuration = TimeSpan.FromSeconds(0.5),
+        OnOpened = async args =>
+        {
+            await Task.Delay(sleepDurationProvider.Current);
+            sleepDurationProvider.MoveNext();
+        }
+
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+**Reasoning**:
+- The lowest value of break duration is half second
+  - That means each sleep is actually takes `sleepDurationProvider.Current` + half second
+- You might think that setting the `BreakDuration` to the `sleepDurationProvider.Current` solves the problem
+  - But it doesn't since the `BreakDuration` is set only once and not re-evalutated at every break
+
+<!-- snippet: circuit-breaker-anti-pattern-2-ext -->
+```cs
+circuitBreaker = new ResiliencePipelineBuilder()
+    .AddCircuitBreaker(new()
+    {
+        ShouldHandle = new PredicateBuilder().Handle<HttpRequestException>(),
+        BreakDuration = sleepDurationProvider.Current,
+        OnOpened = async args =>
+        {
+            Console.WriteLine($"Break: {sleepDurationProvider.Current}");
+            sleepDurationProvider.MoveNext();
+        }
+
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+✅ DO
+
+The `CircuitBreakerStartegyOptions` currently do not provide a way to define break durations dynamically.
```

Comment:
# CircuitBreakerStrategyOptions

The `CircuitBreakerStrategyOptions` currently does not support defining break durations dynamically. This may be re-evaluated in the future. For now, refer to the first example for a potential workaround. However, please use it with caution.